frontend/src/static/js/components/webstatus-overview-table.ts

```diff
@@ -113,6 +117,54 @@ export class WebstatusOverviewTable extends LitElement {
     ];
   }
 
+  findFeaturesFromAtom(
+    searchKey: string,
+    SearchValue: string,
+  ): components['schemas']['Feature'][] {
+    if (!this.taskTracker.data?.data) {
+      return [];
+    }
+
+    const features: components['schemas']['Feature'][] = [];
+    for (const feature of this.taskTracker.data.data) {
+      if (searchKey === 'id' && feature?.feature_id === SearchValue) {
+        features.push(feature);
+        break;
+      } else if (
+        searchKey === 'name' &&
+        (feature?.feature_id.includes(SearchValue) ||
+          feature?.name.includes(SearchValue))
+      ) {
+        features.push(feature);
+      }
+    }
+    return features;
+  }
+
+  sortDataOrder(): components['schemas']['Feature'][] | undefined {
+    if (!this.bookmark || !this.bookmark.is_ordered) {
+      return undefined;
+    }
+
+    const atoms: string[] = this.bookmark.query.trim().split('OR');
```

Comment:
Long term, we should look into integrating with [antlr4's typescript target](https://github.com/antlr/antlr4/blob/master/doc/typescript-target.md) to parse the query (given we can have things like parenthesis and all). We use this same tool to generate Go code already.

Given our bookmarks using this function are simple, we are okay for now. But something to consider in the future.

Reply:
Noted!